pkg/contract/allocations.go

```diff
@@ -235,3 +235,39 @@ func GetEVMSubnetGenesisNativeMinterAdmin(
 	}
 	return getGenesisNativeMinterAdmin(app, network, genesisData)
 }
+
+func ContractAddressIsInBlockchainGenesis(
+	app *application.Avalanche,
+	network models.Network,
+	chainSpec ChainSpec,
+	contractAddress common.Address,
+) (bool, error) {
+	genesisData, err := GetBlockchainGenesis(
+		app,
+		network,
+		chainSpec,
+	)
+	if err != nil {
+		return false, err
+	}
+	if !utils.ByteSliceIsSubnetEvmGenesis(genesisData) {
+		return false, fmt.Errorf("checking for contract belonging to genesis is only supported on EVM based vms")
```

Comment:
nit: error wording is confusing. it seems like `only EVM based vms support genesis contract checks` is better  or some maybe some other way